File: 2644-find-the-maximum-divisibility-score/2644-find-the-maximum-divisibility-score.js
/**
 * @param {number[]} nums
 * @param {number[]} divisors
 * @return {number}
 */
var maxDivScore = function(nums, divisors) {
  const result = [];
  
  for (const divisor of divisors) {
    let count = 0;
    
    for (const num of nums) {
      if (num % divisor === 0) count++;
    }

    const arr = [divisor, count];
    
    result.push(arr);
  }
  
  result.sort((a, b) => {
    if (b[1] !== a[1]) {
      return b[1] - a[1];
    } else {
      return a[0] - b[0];
    }
  })
              
  return result[0][0];
};

File: 2644-find-the-maximum-divisibility-score/2644-find-the-maximum-divisibility-score.py
class Solution:
    def maxDivScore(self, nums: List[int], divisors: List[int]) -> int:
        result = []
        
        for divisor in divisors:
          count = 0
        
          for num in nums:
            if num % divisor == 0: count += 1
          
          result.append([divisor, count])

        result.sort(key = lambda x: (-x[1], x[0]))
        
        return result[0][0]